artman/cli/configure.py

```diff
@@ -46,107 +51,116 @@ def configure(log_level=logging.INFO):
 
     # Go through each step.
     # These are split out to make testing them easier.
-    user_config['local_paths'] = _configure_local_paths(
-        user_config.get('local_paths', {}),
-    )
-    user_config['publish'] = _configure_publish()
-    if user_config['publish'] == 'github':
-        user_config['github'] = _configure_github(
-            user_config.get('github', {}),
-        )
-
-    # Write the final configuration.
-    config_yaml = yaml.dump(user_config,
-        block_seq_indent=2,
-        default_flow_style=False,
-        indent=2,
-    )
-    if isinstance(config_yaml, six.binary_type):
-        config_yaml = config_yaml.decode('utf8')
+    user_config.local.CopyFrom(_configure_local_config())
+    user_config.github.CopyFrom(_configure_github_config())
+
     try:
-        os.makedirs(os.path.expanduser('~/.artman/'))
+        config_dir = os.path.expanduser('~/.artman/')
+        os.makedirs(config_dir)
     except OSError:
         pass
-    with io.open(os.path.expanduser('~/.artman/config.yaml'), 'w+') as file_:
-        file_.write(u'---\n')
-        file_.write(config_yaml)
+    _write_pb_to_yaml(user_config, os.path.join(config_dir, 'config.yaml'))
     logger.success('Configuration written successfully to '
                    '~/.artman/config.yaml.')
 
 
-def _configure_local_paths(local_paths):
-    """Return a copy of user_config with local_paths set.
-
-    Args:
-        local_paths (dict): The starting local_paths portion ofuser config.
+def _configure_local_config():
+    """Determine and return artman user local config.
 
     Returns:
-        dict: The new local_paths dictionary.
+        LocalConfig: The new artman local config settings.
     """
-    answer = copy(local_paths)
-
-    # Ask the user for a repository root.
-    while not answer.get('reporoot'):
-        logger.info('First, we need to know where you store most code on your '
-                    'local machine.')
-        logger.info('Other paths (example: toolkit) will derive from this, '
-                    'but most are individually configurable.')
-        logger.info('The use of ${REPOROOT} in GAPIC YAMLs will point here.')
-        logger.info('Note: Use of ~ is fine here.')
-        answer['reporoot'] = six.moves.input('Local code path: ')
-        answer['reporoot'] = answer['reporoot'].rstrip('/').strip()
-
-    # Set up dependent directories.
-    reporoot = answer['reporoot']
-    for dep in ('api-client-staging', 'googleapis', 'toolkit'):
+    answer = LocalConfig()
+
+    # Ask the user for a local toolkit location.
+    while not answer.toolkit:
         location = six.moves.input(
-            'Path for {0} (default: {1}/{0}): '.format(dep, reporoot)
-        ).rstrip('/').strip()
+            'Where is your local toolkit? Please provide an absolute path: ')
```

Comment:
A new user might not know what "your local toolkit" refers to. Suggest "... your local toolkit repository? (If you do not have a toolkit repository, clone https://github.com/googleapis/toolkit/. Please provide an absolute path: "

Reply:
Accepted the suggestion.